src/mongoose_rabbit_worker.erl

```diff
@@ -0,0 +1,298 @@
+            {noreply, Opts}
+    end.
+
+-spec publish_message_and_wait_for_confirm(Method :: mongoose_amqp:method(),
+                                           Payload :: mongoose_amqp:message(),
+                                           worker_opts()) ->
+    boolean() | timeout | channel_exception.
+publish_message_and_wait_for_confirm(Method, Payload,
+                                     #state{channel = Channel,
+                                            confirms = IsConfirmEnabled}) ->
+    try amqp_channel:call(Channel, Method, Payload) of
+        _Res ->
+            maybe_wait_for_confirms(Channel, IsConfirmEnabled)
+    catch
+        Error:Reason -> {channel_exception, Error, Reason}
+    end.
+
+-spec maybe_wait_for_confirms(Channel :: pid(), boolean()) ->
+    boolean() | timeout.
+maybe_wait_for_confirms(Channel, true) ->
+    amqp_channel:wait_for_confirms(Channel);
+maybe_wait_for_confirms(_, _) -> true.
+
+-spec maybe_restart_rabbit_connection(worker_opts()) -> {pid(), pid()}.
+maybe_restart_rabbit_connection(#state{connection = Conn, channel = Chann,
+                                       host = Host,
+                                       amqp_client_opts = AMQPOpts}) ->
+    case is_process_alive(Conn) of
+        true ->
+            {Conn, amqp_connection:open_channel(Conn)};
+        false ->
+            establish_rabbit_connection(AMQPOpts, Host)
+    end.
+
+-spec establish_rabbit_connection(Opts :: mongoose_amqp:network_params(),
+                                  Host :: jid:server()) -> {pid(), pid()}.
+establish_rabbit_connection(AMQPOpts, Host) ->
+    case amqp_connection:start(AMQPOpts) of
+        {ok, Connection} ->
+            update_success_connections_metrics(Host),
+            {ok, Channel} = amqp_connection:open_channel(Connection),
+            ?DEBUG("event=rabbit_connection_established", []),
+            {Connection, Channel};
+        {error, Error} ->
+            update_failed_connections_metrics(Host),
+            ?ERROR_MSG("event=rabbit_connection_failed reason=~1000p", [Error]),
+            exit("connection to a Rabbit server failed")
+    end.
+
+-spec close_rabbit_connection(Connection :: pid(), Channel :: pid(),
+                              Host :: jid:server()) ->
+    ok | no_return().
+close_rabbit_connection(Connection, Channel, Host) ->
+    update_closed_connections_metrics(Host),
+    try amqp_channel:close(Channel)
+    catch
+        _Error:_Reason -> already_closed
+    end,
+    amqp_connection:close(Connection).
+
+-spec maybe_enable_confirms(Channel :: pid(), proplists:proplist()) ->
+    boolean() | no_return().
+maybe_enable_confirms(Channel, Opts) ->
+    case proplists:get_value(confirms, Opts) of
+        true ->
+            ConfirmCallRes = mongoose_amqp:confirm_select_ok(),
+            ConfirmCallRes =
+                amqp_channel:call(Channel, mongoose_amqp:confirm_select()),
```

Comment:
This `amqp_channel:call` is not guarded by a `try ... catch` like the other ones. I'm not sure if it's important in this case.

Reply:
This is one of worker's initialisation steps (enabling message confirmations from a RabbitMQ server). Should the step fail the worker won't be started.